lib/engine/game/g_1873.rb

```diff
@@ -229,42 +292,86 @@ def place_home_token(corporation)
         end
       end
 
+      # replace railway dummy concession company with a dummy purchase option company
+      def replace_company(corporation)
+        return unless railway?(corporation)
+
+        old_co = @companies.find { |c| c.id == corporation.id }
+        description = "Purchase Option for Railway #{corporation_info[corporation][:name]}"
+        sym = corporation_info[corporation][:sym]
+        name = "#{sym} Purchase Options"
+        @companies[@companies.find_index(old_co)] = Company.new(sym: sym, name: name,
+                                                                value: RAILWAY_MIN_BID, desc: description)
+      end
+
+      def mine?(entity)
+        entity.minor?
+      end
+
+      def independent_mine?(entity)
+        entity.minor? && @corporations.none? { |c| c == entity.owner }
+      end
+
+      def public_mine?(entity)
+        entity.corporation? && @corporation_info[entity][:type] == :mine
```

Comment:
If I'm understanding this right, an entity that is a `public_mine` is not a `mine`? That is, if `public_mine?(entity) == true`, then `mine?(entity) == false`? As someone unfamiliar with Harzbahn that seems confusing--is there a more descriptive name for `mine?` that works?

Reply:
Yes, it probably should be private_mine. I'll clean this up.

Reply:
Actually, there is no need for this method. All private mines are minors.